### PR TITLE
os-programs sub cat added open-source dir #2004

### LINKED ISSUE #2004: [ADD] open source programs subcategory
### Category Name

Open Source

### Subcategory Name

Open Source Programs

### Additional Context

Events like Google Summer of Code, and several notable ones should be added.

### Checklist

- [X] I have checked the existing [issues](https://github.com/rupali-codes/LinksHub/issues?q=is%3Aissue+)
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [ ] I have checked that this category does not exist
- [X] I have checked that this subcategory does not exist
- [ ] I am willing to work on this issue (optional)
- [ ] I am participating in Hacktoberfest 2023

### PR DESCRIPTION
**Pull request by goobric**
## Fixes Issue 

fixes #2004

## Changes proposed
added a sub category 'os programs' to the open-source heading
updated the index.ts, data.ts, CategoryDescriptions.ts, created an os-programs.json file in the open-source dir of the database folder.

## Screenshots
231018_linkshub_os-programs_i2004

## Note to reviewers

@Anmol-Baranwal  thank you for assigning this issue to me :-)

### CODE REVIEW

File: database/open_source/os-programs.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    {
+      "name": "Free Software Foundation",
+      "description": "The Free Software Foundation (FSF) is a nonprofit with a worldwide mission to promote computer user freedom.",
+      "url": "https://www.fsf.org/events/aggregator/",
+      "category": "opensource",
+      "subcategory": "os-programs"
+    },
+    {
+      "name": "OWASP",
+      "description": "To be the global open community that powers secure software through education, tools, and collaboration.",
```
Comment (CBID2) [inline database/open_source/os-programs.json:25]:
```suggestion
      "description": "This is a global open source community that powers secure software through education, tools, and collaboration.",
```

File: database/open_source/os-programs.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      "url": "https://owasp.org/events/",
+      "category": "open-source",
+      "subcategory": "os-programs"
+    },
+    {
+      "name": "The Linux Foundation",
+      "description": "Empowering generations of open source innovators. Unlocking the value of shared technology Innovation, comes from everywhere.",
```
Comment (CBID2) [inline database/open_source/os-programs.json:32]: [suggestion] This description could be improved. Consider using the following questions to help you revise it:
1. What makes Linux Foundation great?
2. Why should open source enthusiasts join the program?
```suggestion
      "description": "Empowering generations of open source innovators. Unlocking the value of shared technology Innovation, comes from everywhere.",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @goobric! :) Your pull request is off to a great start! :) There are some parts that need improvement, so I added feedback on that. :)

**Review by rupali-codes — APPROVED** [review]
lgtm

**Pull request merged**